lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -284,4 +285,38 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
 
         return $result;
     }
+
+    public function listTableDetails($tableName)
+    {
+        $table = parent::listTableDetails($tableName);
+
+        /** @var MySqlPlatform $platform */
+        $platform = $this->_platform;
+        $sql      = $platform->getListTableMetadataSQL($tableName);
+
+        $tableOptions = $this->_conn->fetchAssoc($sql);
+
+        $table->addOption('engine', $tableOptions['ENGINE']);
+        $table->addOption('collation', $tableOptions['TABLE_COLLATION']);
+        if ($tableOptions['AUTO_INCREMENT'] !== '') {
+            $table->addOption('autoincrement', $tableOptions['AUTO_INCREMENT']);
+        }
+        if ($tableOptions['TABLE_COMMENT'] !== '') {
+            $table->addOption('comment', $tableOptions['TABLE_COMMENT']);
+        }
+        if ($tableOptions['CREATE_OPTIONS'] === '') {
+            return $table;
+        }
+
+        $createOptions = [];
+        foreach (explode(' ', $tableOptions['CREATE_OPTIONS']) as $option) {
+            [$createOption, $value] = explode('=', $option);
+
+            $createOptions[$createOption] = $value;
+        }
+
+        $table->addOption('create_options', $createOptions);
```

Comment:
Actually, option are type-hinted as being string. This is an unusual option as it is now an array.

Reply:
It's an invalid hint which will be removed in `develop`. There are existing non-string options:

https://github.com/doctrine/dbal/blob/5140a64c08b4b607b9bedaae0cedd26f04a0e621/lib/Doctrine/DBAL/Platforms/SqlitePlatform.php#L358